cogs/embeds.py

```diff
@@ -1,24 +1,142 @@
+            output = []
+            current = None
+            field_tokens = re.split(" +", field_msg.content.strip())
+
+            for token in field_tokens:
+                if token.startswith("\"") and current is None:
+                    current = ""
+                
+                if current is not None:
+                    current = current + token + " "
+                
+                if token.endswith("\"") and current is not None:
+                    index = len(token) - 2
+                    if index < 0 or token[index] != "\\":
+                        data = current.strip()
+                        data = data[1:-1]
+                        output.append(data)
+                        current = None
+
+            return output
+
+        async def handle_channel():
+            while True:
+                try:
+                    channel_msg: discord.Message = await self.bot.wait_for("message", check=check, timeout=timeout)
+                    if len(channel_msg.channel_mentions) is 0:
+                        await ctx.send("Could not find that channel")
+                    else:
+                        channel = channel_msg.channel_mentions[0]
+                        return channel
+                except TimeoutError:
+                    await ctx.send("Reached Timeout. Discarding Embed")
+                    break
+
+        async def handle_replacement_msg(embed: discord.Embed, channel: discord.TextChannel):
+            msg_id_msg: discord.Message = await self.bot.wait_for("message", check=check, timeout=timeout)
+            msg_id = msg_id_msg.content.strip()
+            try:
+                msg = await channel.fetch_message(msg_id)
+                await msg.edit(embed=embed)
+            except:
+                await channel.send(embed=embed)
+
+        timeout = 60 * 1000
+        embed = discord.Embed()
+        
+        await ctx.send("Please type the description of the embed. Use `skip` to skip this.")
+        desc: discord.Message = await self.bot.wait_for("message", check=check, timeout=timeout)
+        if not is_skip(desc): embed.description = desc.content.strip()[:4096]
+
+        await ctx.send("Please provide the colour in hex format (e.g. `#123456`). Use `skip` to skip this.")
+        await handle_color(embed)
+
+        await ctx.send("Please provide the URL for the image you would like to display. Use `skip` to skip this.")
+        await handle_img(embed)
+
+        await handle_fields(embed)
+        
+        await ctx.send("Which channel should the embed be sent in?")
+        channel = await handle_channel()
+
+        await ctx.send("Should this embed replace any existing message in that channel? If yes, type the message ID, otherwise type anything else.")
+        await handle_replacement_msg(embed, channel)
+
+
     @commands.command()
-    async def embedraw(self, ctx: commands.Context):
+    async def embedraw(self, ctx: commands.Context, *, embed_data):
```

Comment:
`embedraw` should take in a channel ID (where to send the embed) as its first argument.

Reply:
Do we always supply a channel_id in `embedraw`? Even if we want to send the embed in the same channel?

Reply:
Disregard. Found an example and it seems that we do indeed send the channel_id: https://discord.com/channels/230296179991248896/349491266838462465/1049269313401524275